packages/paste-core/components/input/src/Input.tsx

```diff
@@ -72,6 +73,15 @@ export const InputElement = styled.input<InputProps>((props) =>
       '-webkit-text-fill-color': props.variant === 'inverse' ? 'colorTextInverseWeaker' : 'colorTextWeaker',
       '-webkit-opacity': '1',
     },
+    // Fixes extra height issue in Safari
+    '&::-webkit-datetime-edit': {
+      display: 'flex',
+    },
+
+    // Change to pointer cursor on cal icon
+    '&::-webkit-calendar-picker-indicator:hover': {
+      cursor: props.readOnly || props.disabled ? 'default' : 'pointer',
```

Comment:
@richbachman made these changes (along with line 58), just want to get someone else's eyes on it before I merge it in to make sure it's the right approach to solving the cursor fix.